### PR TITLE
Move folding_ranges to ra_ide_api

### PR DESCRIPTION
**Pull request by pasa**
folding_ranges part of #1009

### CODE REVIEW

File: crates/ra_ide_api/src/lib.rs [hunk 1]
```diff
@@ -60,9 +61,10 @@ pub use crate::{
     references::ReferenceSearchResult,
     assists::{Assist, AssistId},
     hover::{HoverResult},
+    folding_ranges::{folding_ranges, Fold, FoldKind},
```
Comment (matklad) [inline crates/ra_ide_api/src/lib.rs]: `folding_ranges` function should be `pub(crate)` I think

Comment (pasa) [reply crates/ra_ide_api/src/lib.rs]: my bad

**Comment by matklad** [comment]
bors r+

Thanks!

**Comment by matklad** [comment]
bors ping

**Comment by matklad** [comment]
bors delegate+

**Pull request merged**